test/test_video.py

```diff
@@ -255,11 +257,11 @@ def _template_read_video(video_object, s=0, e=None):
     audio_frames = torch.empty(0)
     frames = []
     audio_pts = []
-    t, pts = video_object.next()
-    while t.numel() > 0 and (pts > s and pts <= e):
+    for t, pts in itertools.takewhile(lambda x: x[1] <= e, video_object):
+        if pts < s:
```

Comment:
why don't we have 

`for t, pts in itertools.takewhile(lambda x: x[1] <= e, video_object.seek(s))`

to circumvent this line?

Reply:
We already have a `.seek(s)` in line 256. I initially didn't have this condition, but it is all a matter of seeking to the first frame before or after `s`, and `seek` gives us the frame just before `s`, while the code here was doing the opposite.

Ideally I would have preferred to remove this line as well, but IIRC this makes tests fail, so I added it for compatibility. A rewrite of the test suite would be good to have